modem: hl7800: Fix DNS resolver for all address families

rerickson1 opened the pull request:
Fix DNS resolver config for IPv6 only setup.
Check validity of the DNS addresses before trying to configure the resolver.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/54674

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -1829,17 +1829,40 @@ static void dns_work_cb(struct k_work *work)
+#endif
+		NULL};
+
+#ifdef CONFIG_NET_IPV6
+	if (IS_ENABLED(CONFIG_NET_IPV6)) {
```
Why double check, the other one can be removed, preferably the `#ifdef` one

rerickson1 replied:
Need to leave the #ifdef one because `dns_v6_string` is not available if no `CONFIG_NET_IPV6`. I will clean this up.

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -1829,17 +1829,40 @@ static void dns_work_cb(struct k_work *work)
+		}
+	} else
+#endif
+	{
```
So IPv4 is always enabled or why there is no IPv4 check here? Asking because you have a proper check in added code above.

rerickson1 committed:
modem: hl7800: Fix DNS resolver for all address families

Fix DNS resolver config for IPv6 only setup.
Check validity of the DNS addresses before trying to
configure the resolver.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

jukkar reviewed (dismissed):
LGTM

rerickson1 commented:
Rebased to fix CI...
